packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,140 @@
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+  }
+
+  // This method returns the validity of a relay price request attempt. The relay request was valid if and only if it:
+  // The price request's ancillary data contains parameters that match with an L2 deposit event.
+  public async getHistoricalPrice(time: number | string, ancillaryData: string): Promise<BN> {
+    // Note: `time` is unused in this method because it is not included in the relay ancillary data.
+
+    // Parse ancillary data for relay request and find deposit if possible with matching params.
+    const parsedAncillaryData: RelayAncillaryData = (parseAncillaryData(
+      ancillaryData
+    ) as unknown) as RelayAncillaryData;
+    const matchedDeposit = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === parsedAncillaryData.depositId &&
+        deposit.timestamp === parsedAncillaryData.depositTimestamp &&
+        deposit.recipient === toChecksumAddress("0x" + parsedAncillaryData.recipient) &&
+        deposit.sender === toChecksumAddress("0x" + parsedAncillaryData.l2Sender) &&
+        deposit.l1Token === toChecksumAddress("0x" + parsedAncillaryData.l1Token) &&
+        deposit.amount === parsedAncillaryData.amount.toString() &&
+        deposit.slowRelayFeePct === parsedAncillaryData.slowRelayFeePct.toString() &&
+        deposit.instantRelayFeePct === parsedAncillaryData.instantRelayFeePct.toString() &&
+        deposit.quoteTimestamp === parsedAncillaryData.quoteTimestamp &&
+        this.l2Client.bridgeDepositAddress === toChecksumAddress("0x" + parsedAncillaryData.depositContract)
+    );
+
+    // TODO: Do we need to handle the case where all of these params are matched and matchedDeposit.length > 1?
+    if (matchedDeposit.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No deposit event found matching relay request ancillary data and time",
+        parsedAncillaryData,
+      });
+      return toBNWei(isRelayValid.No);
+    }
+
+    // Validate relays proposed realized fee percentage.
+    const expectedRealizedFeePct = this.l1Client.calculateRealizedLpFeesPctForDeposit(/* matchedDeposit[0] */);
+    if (expectedRealizedFeePct !== parsedAncillaryData.realizedLpFeePct.toString()) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "Matched deposit realized fee % is incorrect",
+      });
+      return toBNWei(isRelayValid.No);
+    }
+
+    // Passed all checks, relay is valid!
+    return toBNWei(isRelayValid.Yes);
+  }
+
+  public getLastUpdateTime(): number | null {
+    // TODO.
+    return null;
+  }
+
+  public getLookback(): number | null {
+    // TODO. We could use the L1/L2 client's starting block number and network average block propagation time to
+    // determine this value.
+    return null;
+  }
+
+  public getCurrentPrice(): BN | null {
```

Comment:
it could also error out. when would we be calling this on this feed? seems like something is misconfigured if that was the case